web-registry/src/pages/Ecocredits.tsx

```diff
@@ -0,0 +1,257 @@
+
+const useStyles = makeStyles(theme => ({
+  borderLeft: {
+    // absolutely position border to get around MUI style quirks
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    bottom: 0,
+    borderLeft: `1px solid ${theme.palette.info.light}`,
+  },
+  greyText: {
+    color: theme.palette.info.main,
+  },
+}));
+
+const ROWS_PER_PAGE_OPTIONS = [5, 10];
+
+interface TableCredits extends BatchRowData, EcocreditAccountBalance {}
+
+export const Ecocredits: React.FC = () => {
+  // const walletContext = useWallet();
+  // const wallet = walletContext.wallet?.address;
+  // TODO hard coded for now - the following should work for a user's data
+  const wallet = 'regen1m5fecarvw0ltx2yvvru0kl4un03d3uca2kxggj';
+  const [order, setOrder] = useState<Order>('desc');
+  const [orderBy, setOrderBy] = useState<keyof TableCredits>('start_date');
+  const [credits, setCredits] = useState<TableCredits[]>([]);
+  const { TablePagination, setCountTotal, paginationParams, paginationProps } =
+    useTablePagination(ROWS_PER_PAGE_OPTIONS);
+
+  const fetchData = async (
+    paginationParams: URLSearchParams,
+    setCountTotal: (count: number) => void,
+  ): Promise<void> => {
+    if (!wallet) return;
+    const {
+      data: { batches, pagination },
+    } = await getBatches(paginationParams);
+    const credits = await Promise.all(
+      batches.map(async batch => {
+        const {
+          data: { retired_amount, tradable_amount },
+        } = await getAccontEcocreditsForBatch(batch.batch_denom, wallet);
+        return { ...batch, tradable_amount, retired_amount };
+      }),
+    );
+    setCountTotal(parseInt(pagination.total) || 0);
+    setCredits(credits);
+  };
+
+  useEffect(() => {
+    if (!ledgerRestUri || !paginationParams) return;
+    fetchData(paginationParams, setCountTotal);
+  }, [paginationParams, setCountTotal]);
+
+  const handleSort = (sortBy: keyof TableCredits): void => {
+    const isAsc = orderBy === sortBy && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(sortBy);
+  };
+
+  const SortableHead: React.FC<{ field: string }> = ({ field, children }) => {
+    const isCurrentSort = orderBy === field;
+    return (
+      <StyledTableCell sortDirection={isCurrentSort ? order : false}>
+        <StyledTableSortLabel
+          active={isCurrentSort}
+          direction={isCurrentSort ? order : 'asc'}
+          onClick={() => handleSort(field as keyof TableCredits)}
+        >
+          {children}
+        </StyledTableSortLabel>
+      </StyledTableCell>
+    );
+  };
+
+  const styles = useStyles();
+  return (
+    <Box sx={{ backgroundColor: 'grey.50' }}>
+      <Section title="My Ecocredits" titleVariant="h3" titleAlign="left">
+        <Box
+          sx={{
+            border: 1,
+            borderColor: 'info.light',
+            borderRadius: '8px',
+            marginTop: 8,
+            marginBottom: 12,
+            overflow: 'hidden',
```

Comment:
I prioritized using `sx` for one-off styles and only using `makeStyles` for stuff that is reused